browser/ui/views/tabs/brave_tab_strip.cc

```diff
@@ -161,7 +162,23 @@ void BraveTabStrip::MaybeStartDrag(
     }
   }
 
-  TabStrip::MaybeStartDrag(source, event, original_selection);
+  auto new_selection = original_selection;
+  if (source->GetTabSlotViewType() == TabSlotView::ViewType::kTab) {
+    auto* tab = static_cast<Tab*>(source);
+    if (auto tile = GetTileForTab(tab)) {
+      // Make a pair of tabs in a tile selected together so that they move
+      // together during drag and drop.
+      auto* tab_strip_model = controller_->GetBrowser()->tab_strip_model();
+      auto index = tab_strip_model->GetIndexOfTab(
+          IsFirstTabInTile(tab) ? tile->second : tile->first);
+      DCHECK_NE(index, TabStripModel::kNoTab);
+      new_selection.AddIndexToSelection(index);
```

Comment:
Hey, so it looks like the selection is stored in a `flat_set` - instead of working out which tab is selected and selecting the other one we could just add both?

```cpp
// Ensure both tabs are selected
AddIndexToSelection(first);
AddIndexToSelection(second);

tsm->SetSelectionModel(...)
```

Reply:
Yeah that could be an option, but I slightly prefer the way as is. If we added both of them, the code will be like

```
      new_selection.AddIndexToSelection(tab_strip_model->GetIndexOfTab(tile->first));
      new_selection.AddIndexToSelection(tab_strip_model->GetIndexOfTab(tile->second));
```

And I don't see big benefit from this over the current code, especially when we can find out what is missing.

Reply:
Hey @fallaciousreasoning, you were right. I got crash due to not having the selected tab in a certain scenario. Not sure why 😂 . Will change right away

Reply:
Huh, that's kinda weird :laughing: